1.11.0-rc1 Panic: runtime error: invalid memory address or nil pointer dereference

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Running policies:
```bash
k get clusteradmissionpolicies.policies.kubewarden.io                                                                                                0.006s
NAME                        POLICY SERVER   MUTATING   BACKGROUNDAUDIT   MODE      OBSERVED MODE   STATUS   AGE
psa-label-enforcer-policy   default         true       true              protect   protect         active   9m2s
privileged-pods             default         false      true              protect   protect         active   9m44s
safe-labels                 default         false      true              protect   protect         active   8m19s
```

Error:
```
{"level":"debug","ca-cert-file":"/pki/policy-server-root-ca-pem","time":"2024-03-01T12:03:15Z","message":"appended cert file to in-app RootCAs trust store"}
{"level":"info","time":"2024-03-01T12:03:15Z","message":"clusterwide resources scan started"}
{"level":"info","dict":{"policies to evaluate":2,"policies skipped":0},"time":"2024-03-01T12:03:15Z","message":"cluster admission policies count"}
{"level":"debug","response":{"uid":"7e8491fa-c84e-4e55-aad0-45c0dde2ddf2","allowed":true,"policy":"psa-label-enforcer-policy","resource":"kube-system"},"time":"2024-03-01T12:03:15Z","message":"audit review response"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x68 pc=0x15d144f]

goroutine 1 [running]:
github.com/kubewarden/audit-scanner/internal/report.newPolicyReportResult({0x1c8a0d8, 0xc000478200}, 0xc00036d440, 0x0, {0xc000250ee0?, 0x0?})
	/workspace/internal/report/report.go:136 +0x8f
github.com/kubewarden/audit-scanner/internal/report.AddResultToClusterPolicyReport(0xc0006d0580, {0x1c8a0d8, 0xc000478200}, 0xc000250ee0?, 0xb?)
	/workspace/internal/report/report.go:114 +0x9a
github.com/kubewarden/audit-scanner/internal/scanner.(*Scanner).auditClusterResource(0xc0006b8370, {0x1c73cc0, 0x295a5e0}, {0xc0002b87c0, 0x2, 0xc000279660?}, {0xc00051e090})
	/workspace/internal/scanner/scanner.go:309 +0x4f3
github.com/kubewarden/audit-scanner/internal/scanner.(*Scanner).ScanClusterWideResources.func1({0x1c60a30?, 0xc000132f40?})
	/workspace/internal/scanner/scanner.go:198 +0x4b
k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*UnstructuredList).EachListItem(0xc0002865c0, 0xc00041c8a0)
	/go/pkg/mod/k8s.io/apimachinery@v0.29.2/pkg/apis/meta/v1/unstructured/unstructured_list.go:48 +0x6d
k8s.io/apimachinery/pkg/api/meta.eachListItem({0x1c609e0?, 0xc0002865c0}, 0xc00041c8a0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.29.2/pkg/api/meta/help.go:136 +0x9c
k8s.io/apimachinery/pkg/api/meta.EachListItem(...)
	/go/pkg/mod/k8s.io/apimachinery@v0.29.2/pkg/api/meta/help.go:119
k8s.io/client-go/tools/pager.(*ListPager).EachListItem.func1({0x1c609e0?, 0xc0002865c0?})
	/go/pkg/mod/k8s.io/client-go@v0.29.2/tools/pager/pager.go:185 +0x25
k8s.io/client-go/tools/pager.(*ListPager).eachListChunkBuffered(0xc0000cc468, {0x1c73cc0, 0x295a5e0}, {{{0x0, 0x0}, {0x0, 0x0}}, {0x0, 0x0}, {0x0, ...}, ...}, ...)
	/go/pkg/mod/k8s.io/client-go@v0.29.2/tools/pager/pager.go:245 +0x2bb
k8s.io/client-go/tools/pager.(*ListPager).EachListItem(0xc00025ba60?, {0x1c73cc0?, 0x295a5e0?}, {{{0x0, 0x0}, {0x0, 0x0}}, {0x0, 0x0}, {0x0, ...}, ...}, ...)
	/go/pkg/mod/k8s.io/client-go@v0.29.2/tools/pager/pager.go:184 +0x85
github.com/kubewarden/audit-scanner/internal/scanner.(*Scanner).ScanClusterWideResources(0xc0006b8370, {0x1c73cc0?, 0x295a5e0})
	/workspace/internal/scanner/scanner.go:192 +0x466
github.com/kubewarden/audit-scanner/cmd.startScanner({0x0?, 0xc0000fb180?}, 0x10?, 0x0?)
	/workspace/cmd/root.go:125 +0x9b
github.com/kubewarden/audit-scanner/cmd.glob..func1(0xc0001b8b00?, {0x19f7fae?, 0x4?, 0x19f7f76?})
	/workspace/cmd/root.go:92 +0x477
github.com/spf13/cobra.(*Command).execute(0x29156a0, {0xc000134010, 0x52, 0x57})
	/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0xabc
github.com/spf13/cobra.(*Command).ExecuteC(0x29156a0)
	/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/kubewarden/audit-scanner/cmd.Execute()
	/workspace/cmd/root.go:103 +0x36
main.main()
	/workspace/main.go:8 +0xf
```

### To reproduce:

```bash
# resources/default-kubewarden-controller-values.yaml
telemetry:
  enabled: false
policyServer:
  env:
    - name: KUBEWARDEN_LOG_LEVEL
      value: debug
auditScanner:
  enable: true
  image:
    tag: v1.11.0-rc1
image:
  tag: v1.11.0-rc1


# resources/default-kubewarden-defaults-values.yaml
recommendedPolicies:
  enabled: False
policyServer:
  replicaCount: 1
  image:
    tag: v1.11.0-rc1

# remove --store from controller chart
helm pull --repo https://charts.kubewarden.io/ kubewarden-controller --untar
sed -i '/store/d' ~/kubewarden-controller/templates/_helpers.tpl.txt

# Change Makefile to use local chart
-               $(KUBEWARDEN_CONTROLLER_CHART_RELEASE) $(KUBEWARDEN_CHARTS_LOCATION)/kubewarden-controller
+               $(KUBEWARDEN_CONTROLLER_CHART_RELEASE) ./kubewarden-controller


# Open tests/audit-scanner.bats
# comment out teardown function

# Run tests
make clean cluster install audit-scanner.bats
kubectl create job  --from=cronjob/audit-scanner testing  --namespace kubewarden
kubectl get pods -n kubewarden
```

--- a/internal/report/report.go
+++ b/internal/report/report.go
@@ -132,7 +132,11 @@ func newPolicyReportResult(policy policiesv1.Policy, admissionReview *admissionv
 	}
 
 	var message string
-	if !errored {
+	// We need to check if Result is not nil because this field is
+	// optional. If the policy returns "allowed" to the admissionReview,
+	// the Result field is not checked by Kubernetes.
+	// https://pkg.go.dev/k8s.io/api@v0.29.2/admission/v1#AdmissionResponse
+	if !errored && admissionReview.Response.Result != nil {
 		message = admissionReview.Response.Result.Message
 	}
 

--- a/internal/report/report_test.go
+++ b/internal/report/report_test.go
@@ -235,6 +235,46 @@ func TestNewPolicyReportResult(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "Validating policy, allowed response with no message",
+			policy: &policiesv1.ClusterAdmissionPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					UID:             "policy-uid",
+					ResourceVersion: "1",
+					Name:            "policy-name",
+					Annotations: map[string]string{
+						policiesv1.AnnotationSeverity: severityLow,
+					},
+				},
+				Spec: policiesv1.ClusterAdmissionPolicySpec{
+					PolicySpec: policiesv1.PolicySpec{
+						Mutating: false,
+					},
+				},
+			},
+			amissionReview: &admissionv1.AdmissionReview{
+				Response: &admissionv1.AdmissionResponse{
+					Allowed: true,
+					Result:  nil,
+				},
+			},
+			errored: false,
+			expectedResult: &wgpolicy.PolicyReportResult{
+				Source:          policyReportSource,
+				Policy:          "clusterwide-policy-name",
+				Severity:        severityLow,
+				Result:          statusPass,
+				Timestamp:       now,
+				Scored:          true,
+				SubjectSelector: &metav1.LabelSelector{},
+				Description:     "",
+				Properties: map[string]string{
+					PropertyPolicyUID:             "policy-uid",
+					propertyPolicyResourceVersion: "1",
+					typeValidating:                valueTypeTrue,
+				},
+			},
+		},
 	}
 
 	for _, test := range tests {

--- a/internal/report/report_test.go
+++ b/internal/report/report_test.go
@@ -137,7 +137,7 @@ func TestNewPolicyReportResult(t *testing.T) {
 			amissionReview: &admissionv1.AdmissionReview{
 				Response: &admissionv1.AdmissionResponse{
 					Allowed: true,
-					Result:  &metav1.Status{Message: "The request was allowed"},
+					Result:  nil,
 				},
 			},
 			errored: false,
@@ -149,7 +149,7 @@ func TestNewPolicyReportResult(t *testing.T) {
 				Timestamp:       now,
 				Scored:          true,
 				SubjectSelector: &metav1.LabelSelector{},
-				Description:     "The request was allowed",
+				Description:     "",
 				Properties: map[string]string{
 					PropertyPolicyUID:             "policy-uid",
 					propertyPolicyResourceVersion: "1",
@@ -224,46 +224,6 @@ func TestNewPolicyReportResult(t *testing.T) {
 				Policy:          "namespaced-policy-namespace-policy-name",
 				Severity:        severityInfo,
 				Result:          statusError,
-				Timestamp:       now,
-				Scored:          true,
-				SubjectSelector: &metav1.LabelSelector{},
-				Description:     "",
-				Properties: map[string]string{
-					PropertyPolicyUID:             "policy-uid",
-					propertyPolicyResourceVersion: "1",
-					typeValidating:                valueTypeTrue,
-				},
-			},
-		},
-		{
-			name: "Validating policy, allowed response with no message",
-			policy: &policiesv1.ClusterAdmissionPolicy{
-				ObjectMeta: metav1.ObjectMeta{
-					UID:             "policy-uid",
-					ResourceVersion: "1",
-					Name:            "policy-name",
-					Annotations: map[string]string{
-						policiesv1.AnnotationSeverity: severityLow,
-					},
-				},
-				Spec: policiesv1.ClusterAdmissionPolicySpec{
-					PolicySpec: policiesv1.PolicySpec{
-						Mutating: false,
-					},
-				},
-			},
-			amissionReview: &admissionv1.AdmissionReview{
-				Response: &admissionv1.AdmissionResponse{
-					Allowed: true,
-					Result:  nil,
-				},
-			},
-			errored: false,
-			expectedResult: &wgpolicy.PolicyReportResult{
-				Source:          policyReportSource,
-				Policy:          "clusterwide-policy-name",
-				Severity:        severityLow,
-				Result:          statusPass,
 				Timestamp:       now,
 				Scored:          true,
 				SubjectSelector: &metav1.LabelSelector{},

--- a/internal/scanner/scanner_test.go
+++ b/internal/scanner/scanner_test.go
@@ -33,9 +33,7 @@ func newMockPolicyServer() *httptest.Server {
 		admissionReview := admissionv1.AdmissionReview{
 			Response: &admissionv1.AdmissionResponse{
 				Allowed: true,
-				Result: &metav1.Status{
-					Message: "The request was allowed",
-				},
+				Result:  nil,
 			},
 		}
 		response, err := json.Marshal(admissionReview)

